docs/config.rst

```diff
@@ -1,11 +1,505 @@
+
+
+
+Using Tasks and Flows From a Different Project
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+With CumulusCI, it's also possible to use arbitrary tasks and flows from another project.
+To do this, the other project must be named in the ``sources`` section of the ``cumulusci.yml``:
+
+.. code-block:: yaml
+
+    sources:
+      npsp:
+        github: https://github.com/SalesforceFoundation/NPSP
+
+This says that when tasks or flows are referenced using the `npsp` namespace, CumulusCI should fetch the source from this GitHub repository.
+By default, it will fetch the most recent release, or the default branch if there are no releases.
+It's also possible to fetch a specific ``tag``:
+
+.. code-block:: yaml
+
+    sources:
+      npsp:
+        github: https://github.com/SalesforceFoundation/NPSP
+        tag: rel/3.163
+
+or a specific ``commit`` or ``branch``.
+
+Now it's possible to run a flow from NPSP:
+
+.. code-block:: console
+
+    $ cci flow run npsp:install_prod
+
+Or a task:
+
+.. code-block:: console
+
+    $ cci task run npsp:robot
+
+Or even to create a new flow which uses a flow from NPSP:
+
+.. code-block:: yaml
+
+    flows:
+      install_npsp:
+        steps:
+          1:
+            flow: npsp:install_prod
+          2:
+            flow: dev_org
+
+This flow will use NPSP's ``install_prod`` flow to install NPSP as a managed package, and then run this project's own ``dev_org`` flow.
+
+
+
+Troubleshooting Configurations
+------------------------------
+Use ``cci task info <task_name>`` and ``cci flow info <flow_name>`` to see how a given task or flow will behave with the current state of configuration.
+For example, the ``util_sleep`` task has a ``seconds`` option that defaults to 5 out-of-the-box:
+
+.. code-block:: console
+
+    $ cci task info util_sleep
+    util_sleep
+
+    Description: Sleeps for N seconds
+
+    Class: cumulusci.tasks.util.Sleep
+
+    Command Syntax
+
+        $ cci task run util_sleep
+
+    Options
+
+        -o seconds SECONDS
+        Required
+        The number of seconds to sleep
+        Default: 5
+
+If you instead want the default value to be 30 seconds for all projects you could add the following in your global ``cumulusci.yml`` file locaated at ``~/.cumulusci/cumulusci.yml``:
+
+.. code-block:: yaml
+
+    tasks:
+        util_sleep:
+            options:
+                seconds: 30
+
+And now ``cci task info util_sleep`` shows a default of 30 seconds:
+
+.. code-block:: console
+
+    $ cci task info util_sleep
+    util_sleep
+
+    Description: Sleeps for N seconds
+
+    Class: cumulusci.tasks.util.Sleep
+
+    Command Syntax
+
+        $ cci task run util_sleep
+
+    Options
+
+        -o seconds SECONDS
+        Required
+        The number of seconds to sleep
+        Default: 30
+
+Being able to display the active configuration for a given task or flow can help with cross-correlating which configuration scope is affecting a specific scenario.
```

Comment:
I could imagine creating a command for debugging configuration that actually prints out where each line in the merged config came from.

Reply:
`cci task info <task_name> --debug-config`

Reply:
Except I was thinking for all of cumulusci.yml, not just tasks